File: disappearing_num.py
""" Find All Numbers Disappeared in an Array 
Time Complexity : O(n)
Space Complexity: O(1)
aproach : Inplace modification 
"""
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        for i in range(len(nums)):
            new_index = abs(nums[i])-1
            #take the value at nums[i] as new index
            if nums[new_index]>0:
                nums[new_index] =nums[new_index]*(-1) #change the element in the new index to negative -> to show that index (position) as visited
       
        result = []
        for i in range(1,len(nums)+1):
            if nums[i-1]>0:
                result.append(i)
        return result

File: DisappearingNum.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None

import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {

        List<Integer> result = new ArrayList<>();

        // Based on the num encountered Mark the corresponding index as visited by
        // making it negative
        for (int i = 0; i < nums.length; i++) {
            int idx = Math.abs(nums[i]) - 1;
            if (nums[idx] > 0) {
                nums[idx] = nums[idx] * -1;
            }
        }

        // Iterate the array again and check if there are any positive numbers remaining
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                result.add(i + 1);
            }
        }

        return result;
    }
}